velox/functions/prestosql/Sequence.cpp

```diff
@@ -0,0 +1,173 @@
+
+namespace facebook::velox::functions {
+namespace {
+
+// See documentation at https://prestodb.io/docs/current/functions/array.html
+class SequenceFunction : public exec::VectorFunction {
+ public:
+  static constexpr int32_t kMaxResultEntries = 10'000;
+
+ private:
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    VectorPtr localResult;
+    try {
+      if (args[0]->type() == BIGINT()) {
+        localResult = applyNumber(rows, args, outputType, context);
+      } else {
+        VELOX_FAIL("Unsupported type for argument of 'sequence' function");
+      }
+    } catch (const std::exception& e) {
+      context.setErrors(rows, std::current_exception());
+      return;
+    }
+    context.moveOrCopyResult(localResult, rows, result);
+  }
+
+  VectorPtr applyNumber(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context) const {
+    const auto numRows = rows.size();
+    auto pool = context.pool();
+    vector_size_t numElements = 0;
+
+    exec::DecodedArgs decodedArgs(rows, args, context);
+    auto startVector = decodedArgs.at(0);
+    auto stopVector = decodedArgs.at(1);
+    DecodedVector* stepVector = nullptr;
+    if (args.size() == 3) {
+      stepVector = decodedArgs.at(2);
+    }
+
+    context.applyToSelectedNoThrow(rows, [&](auto row) {
+      numElements += sequenceCount(startVector, stopVector, stepVector, row);
+    });
+
+    VectorPtr elements = BaseVector::create(BIGINT(), numElements, pool);
+    BufferPtr sizes = allocateSizes(numRows, pool);
+    BufferPtr offsets = allocateOffsets(numRows, pool);
+    auto rawElements = elements->asFlatVector<int64_t>()->mutableRawValues();
+    auto rawSizes = sizes->asMutable<vector_size_t>();
+    auto rawOffsets = offsets->asMutable<vector_size_t>();
+
+    vector_size_t elementsOffset = 0;
+    context.applyToSelectedNoThrow(rows, [&](auto row) {
+      rawOffsets[row] = elementsOffset;
+      rawSizes[row] = sequenceCount(startVector, stopVector, stepVector, row);
+      writeToElements(
+          rawElements,
+          elementsOffset,
+          startVector,
+          stopVector,
+          stepVector,
+          row);
+      elementsOffset += rawSizes[row];
+    });
+
+    return std::make_shared<ArrayVector>(
+        pool, outputType, nullptr, numRows, offsets, sizes, elements);
+  }
+
+  vector_size_t sequenceCount(
+      DecodedVector* startVector,
+      DecodedVector* stopVector,
+      DecodedVector* stepVector,
+      vector_size_t row) const {
+    auto start = startVector->valueAt<int64_t>(row);
+    auto stop = stopVector->valueAt<int64_t>(row);
+    int64_t step;
+    if (stepVector == nullptr) {
+      step = stop >= start ? 1 : -1;
+    } else {
+      step = stepVector->valueAt<int64_t>(row);
+    }
```

Comment:
Leaving uninitialized variable can later lead to bugs if code after it is changed.
We can leverage ? operator here:
```
  const int64_t step = (stepVector == nullptr)
      ? (stop >= start ? 1 : -1)
      : step = stepVector->valueAt<int64_t>(row);
```

or 

```
int64_t step{0};
if() else ();
```